mps-sync-plugin/src/main/kotlin/org/modelix/mps/sync/plugin/action/ModelixActionGroup.kt

```diff
@@ -22,24 +22,48 @@ import com.intellij.openapi.actionSystem.AnActionEvent
 import org.modelix.kotlin.utils.UnstableModelixFeature
 import org.modelix.mps.sync.plugin.icons.CloudIcons
 
-@UnstableModelixFeature(reason = "The new modelix MPS plugin is under construction", intendedFinalization = "This feature is finalized when the new sync plugin is ready for release.")
+@UnstableModelixFeature(
+    reason = "The new modelix MPS plugin is under construction",
+    intendedFinalization = "This feature is finalized when the new sync plugin is ready for release.",
+)
 class ModelixActionGroup : ActionGroup("Modelix Actions", "", CloudIcons.PLUGIN_ICON) {
 
+    private val modelActions = arrayOf(ModelSyncAction, UnbindModelAction)
+    private val moduleActions = arrayOf(ModuleSyncAction, UnbindModuleAction)
+
     init {
         isPopup = true
     }
 
     override fun getChildren(event: AnActionEvent?): Array<AnAction> {
-        val model = event?.getData(ModelSyncAction.CONTEXT_MODEL)
+        val model = event?.getData(ModelSyncAction.contextModel)
         if (model != null) {
-            return arrayOf(ModelSyncAction.create(), UnbindModelAction.create())
+            return modelActions
         }
 
-        val module = event?.getData(ModuleSyncAction.CONTEXT_MODULE)
+        val module = event?.getData(ModuleSyncAction.contextModule)
         if (module != null) {
-            return arrayOf(ModuleSyncAction.create(), UnbindModuleAction.create())
+            return moduleActions
         }
 
         return emptyArray()
     }
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
Could this class be made a `data class` to avoid custom `.equals` and `.hashCode`?

Reply:
We can do that, but then `ModelixActionGroup` must have a "dummy" constructor argument.

If we make `modelActions` and `moduleActions` then it's recommended `ModelixActionGroup` has `hashCode` and `equals` (recommendation is coming from intelliJ). Which means, we do not move away from `equals` and `hashCode`. 

Third option is to leave everything as is.

Which way would you like to take?

Reply:
fixed in f6d88799131dd93b5cd95b8c6af6ace5e4c75863